packages/itwin/tree-widget/src/test/trees/VisibilityTreeEventHandler.test.ts

```diff
@@ -36,124 +48,131 @@ describe("VisibilityTreeEventHandler", () => {
     dispose: sinon.fake(),
   };
 
-  const testVisibilityStatus: VisibilityStatus = {
-    state: "visible",
-    isDisabled: false,
-  };
+  before(async () => {
+    await NoRenderApp.startup();
+    await TestUtils.initialize();
+  });
 
-  beforeEach(() => {
-    getVisibilityStatus.returns(testVisibilityStatus);
-    modelMock.setup((x) => x.getNode(moq.It.isAny())).returns(() => createSimpleTreeModelNode());
-    modelSourceMock.setup((x) => x.onModelChanged).returns(() => new BeUiEvent<[TreeModel, TreeModelChanges]>());
-    modelSourceMock.setup((x) => x.getModel()).returns(() => modelMock.object);
-    nodeLoaderMock.setup((x) => x.dataProvider).returns(() => dataProviderMock.object);
-    nodeLoaderMock.setup((x) => x.modelSource).returns(() => modelSourceMock.object);
+  after(async () => {
+    TestUtils.terminate();
+    await IModelApp.shutdown();
   });
 
   afterEach(() => {
-    modelSourceMock.reset();
-    modelMock.reset();
-    nodeLoaderMock.reset();
-    dataProviderMock.reset();
-    selectionHandlerMock.reset();
-    getVisibilityStatus.reset();
     changeVisibility.reset();
+    onVisibilityChange.clear();
+    getVisibilityStatus.resetHistory();
+    sinon.restore();
   });
 
+  function setupTreeModel(nodeIds: string[], item?: PresentationTreeNodeItem) {
+    const model = new MutableTreeModel();
+    model.setChildren(
+      undefined,
+      nodeIds.map((nodeId) => {
+        return { ...createSimpleTreeModelNode(nodeId), isLoading: false, item: item ?? createSimpleTreeModelNode("node-item") };
+      }),
+      0,
+    );
+    const modelSource = new TreeModelSource(model);
+
+    nodeLoaderStub = {
+      loadNode: sinon.stub<Parameters<ITreeNodeLoader["loadNode"]>, ReturnType<ITreeNodeLoader["loadNode"]>>(),
```

Comment:
Does not look like `loadNode` is configured in any of the tests. Maybe change it to () => EMPTY?